File: src/main/kotlin/raft/war/binary/parser/w3g/parser/replay/PlayerRecord.java
package raft.war.binary.parser.w3g.parser.replay;

import raft.war.binary.parser.w3g.parser.packed.IRecord;
import raft.war.binary.parser.w3g.parser.utils.PlayerInfo;

import java.nio.ByteBuffer;

public class PlayerRecord extends PlayerInfo implements IRecord {

    public static final int TYPE = 0x16;
    private int unknown;

    @Override
    public int getRecordId() {
        return TYPE;
    }

    @Override
    public void parse(ByteBuffer inBuffer) {
        super.parse(inBuffer);

        unknown = inBuffer.getInt();
    }

    @Override
    public ByteBuffer assembly(ByteBuffer outBuffer) {
        ByteBuffer playerInfoBytes = super.assembly(null).flip();

        if (outBuffer == null)
            outBuffer = ByteBuffer.allocate(playerInfoBytes.remaining() + 4);

        outBuffer.put(playerInfoBytes);
        outBuffer.putInt(unknown);

        return outBuffer;
    }

    public int getUnknown() {
        return unknown;
    }

    public PlayerRecord setUnknown(int unknown) {
        this.unknown = unknown;
        return this;
    }


}

File: src/main/kotlin/raft/war/binary/parser/w3g/record/PlayerRecord.kt
package raft.war.binary.parser.w3g.record

import java.nio.ByteBuffer

class PlayerRecord : PlayerRecordData(), RecordBase {
    override var timestamp: Long = 0

    var unknown: Int = 0

    override fun parse(inBuffer: ByteBuffer) {
        super.parse(inBuffer)
        unknown = inBuffer.getInt()
    }

    companion object {
        const val ID: Int = 0x16
    }

    override fun toString(): String = "☠️Time: $timestamp,\tPlayerRecord\n" + super.toString() +
            "\tunknown: $unknown\n"
}
